compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -923,7 +923,13 @@ pub fn version_at_macro_invocation(
 
         let debug_flags = matches.opt_strs("Z");
         let backend_name = debug_flags.iter().find_map(|x| x.strip_prefix("codegen-backend="));
-        get_codegen_backend(handler, &None, backend_name).print_version();
+        let sopts = config::Options::default();
+        let sysroot = sopts.maybe_sysroot.clone().unwrap_or_else(|| {
+            filesearch::get_or_default_sysroot().expect("Failed finding sysroot")
+        });
+        let target = config::build_target_config(handler, &sopts, None, &sysroot);
```

Comment:
This snippet appears several times. Any way to refactor it into the normal code path?

Reply:
Idk, the crate structure and everything is very confusing, so I'm not sure how to refactor this...

I've moved the maybe_sysroot->sysroot into a function, but I don't think I can do much more